File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataDao.java
package com.cajetan.youtubeplus.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface VideoDataDao {

    // TODO: implement some ordering

    @Query("SELECT * FROM video_data_table")
    LiveData<List<VideoData>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(VideoData... users);

    @Delete
    void delete(VideoData data);
}

File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataDao.kt
package com.cajetan.youtubeplus.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface VideoDataDao {
    // TODO: implement some ordering

    @Query("SELECT * FROM video_data_table")
    fun getAll(): LiveData<List<VideoData>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(vararg users: VideoData)

    @Query("SELECT * FROM video_data_table WHERE video_id = :id LIMIT 1")
    fun getWithId(id: String): VideoData?

    @Delete
    fun delete(data: VideoData)
}